header-only mode: add CGALConfigVersion.cmake

lrineau opened the pull request:
## Summary of Changes

- Add a file `CGALConfigVersion.cmake` next to the header-only `CGALConfig.cmake`. That allows to add a minimal version of CGAL, as in `find_package(CGAL 4.14.2 REQUIRED COMPONENTS Core)`.
- Modify the release creation script to take that file as input for the release version numbers, instead of `Maintenance/release_building/`

## Release Management 
* Fix bug #4720
* Affected package(s): Installation, Scripts
* License and copyright ownership: maintenance by GF

lrineau commented:
There are also five CTest test, in `test/Installation/`: they can be launched with:

    ctest -R test_find_package_version

lrineau commented:
In my opinion, this PR is a release-blocker.

@sloriot What do you think?

sloriot commented on Installation/test/Installation/test_find_package.cmake.in:
```diff
@@ -0,0 +1,14 @@
+cmake_minimum_required(VERSION ${CMAKE_VERSION})
+project( test_find_package_${mode} )
+find_package(CGAL ${VERSION} ${EXACT}REQUIRED
```
Is there a missing space after EXACT?

lrineau replied:
No: `${EXACT}` evaluates to an empty string, or to `EXACT ` (with a space).

But maybe I should add the space, for readability. That does not change anything.

sloriot commented on Scripts/developer_scripts/create_new_release:
```diff
@@ -180,9 +180,13 @@ else
+    MINOR_NUMBER=`cat ${NUMBERS_DIR}/MINOR_NUMBER`     # 2 digits max
+    BUGFIX_NUMBER=`cat ${NUMBERS_DIR}/BUGFIX_NUMBER`   # 1 digit max
+else
+    eval $(cmake -DCGALCONFIGVERSIONFILE=${SOURCES_DIR}/CGALConfigVersion.cmake -P ${SOURCES_DIR}/Scripts/developer_scripts/create_new_release_evaluate_versions.cmake 2>&1)
```
Why is an include not sufficient?

lrineau replied:
`create_new_release` is a shell script and `CGALConfigVersion.cmake` is a CMake script. They do no share any part of common syntax.

sloriot replied:
Sorry, I thought it was the new cmake script to create a release. BTW, I thought that the cmake script was the one currently in use.

lrineau replied:
We actually have three scripts:
- the CMake script `Scripts/developer_scripts/cgal_create_release_with_cmake.cmake` creates an internal release, or a public release, from a Git checkout,
- the shell script `Scripts/developer_scripts/create_new_release` creates both the internal release, the public release, and the Windows installer, and it calls the previous one,
- the shell script `Maintenance/infrastructure/cgal.geometryfactory.com/bin/create_release` is for me only: it hard-codes local Git repositories, do the `git pull`/`git push`, calls the previous script, and then public the tarballs and installer on the web site.

maxGimeno commented:
This needs a maser version, it changes files that do not exist anymore.

lrineau commented:
> This needs a maser version, it changes files that do not exist anymore.

The merge conflicts is:
```
Unmerged paths:
  (use "git add/rm <file>..." as appropriate to mark resolution)
        deleted by them: Maintenance/release_building/BUGFIX_NUMBER
        deleted by them: Maintenance/release_building/MAJOR_NUMBER
        deleted by them: Maintenance/release_building/MINOR_NUMBER
        deleted by them: Maintenance/release_building/public_release_name
```
because this PR **removes** the file. I explained it [in the first message](https://github.com/CGAL/cgal/pull/4697#issue-412163826). So the conflict resolution is to remove those files as well.

Actually, there is also another conflict that is less clear: the numbers in `Installation/lib/cmake/CGAL/CGALConfigVersion.cmake` must be adapted to what is in `Installation/include/CGAL/version.h`.

lrineau commented:
I have tried to merge the PR in `integration` (1bec158b4a154deb1a6fcedde9f09e560e78df25), but there are already errors in previous merges in `integration`. `Point_set_processing_3/examples/Point_set_processing_3/CMakeLists.txt` defines several targets several times.

maxGimeno commented:
This PR creates a "UNKNOWN LABEL" line on platforms that use ctest : https://cgal.geometryfactory.com/CGAL/testsuite/results-5.1-Ic-149.shtml#UNKNOWN_LABEL

lrineau commented:
> This PR creates a "UNKNOWN LABEL" line on platforms that use ctest : https://cgal.geometryfactory.com/CGAL/testsuite/results-5.1-Ic-149.shtml#UNKNOWN_LABEL

I just fix that in https://github.com/CGAL/cgal/pull/4697/commits/7d64c59b860d73d4cb8a74f038e979cc17309060 and another bug in https://github.com/CGAL/cgal/pull/4697/commits/d30dfacf5fefa4ff2dc8552be77f0adc3e4232d5.

lrineau commented:
I have reworked this PR. It will fix also the issue #4433.

maxGimeno commented:
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.1-Ic-156.shtml
